spring-integration-zeromq/src/main/java/org/springframework/integration/zeromq/channel/ZeroMqChannel.java

```diff
@@ -0,0 +1,315 @@
+								.doOnNext((port) ->
+										setConnectUrl("tcp://localhost:" + this.zeroMqProxy.getFrontendPort() +
+												':' + this.zeroMqProxy.getBackendPort()));
+					}
+					else {
+						return Mono.empty();
+					}
+				})
+						.cache();
+
+		this.sendSocket =
+				proxyMono
+						.publishOn(this.publisherScheduler)
+						.then(Mono.fromCallable(() ->
+								this.context.createSocket(
+										this.connectSendUrl == null
+												? SocketType.PAIR
+												: (this.pubSub ? SocketType.XPUB : SocketType.PUSH))
+						))
+						.doOnNext(this.sendSocketConfigurer)
+						.doOnNext((socket) ->
+								socket.connect(this.connectSendUrl != null
+										? this.connectSendUrl
+										: localPairConnection.get()))
+						.delayUntil((socket) ->
+								(this.pubSub && this.connectSendUrl != null)
+										? Mono.just(socket).map(ZMQ.Socket::recv)
+										: Mono.empty())
+						.cache();
+
+		this.subscribeSocket =
+				proxyMono
+						.publishOn(this.subscriberScheduler)
+						.then(Mono.fromCallable(() ->
+								this.context.createSocket(
+										this.connectSubscribeUrl == null
+												? SocketType.PAIR
+												: (this.pubSub ? SocketType.SUB : SocketType.PULL))))
+						.doOnNext(this.subscribeSocketConfigurer)
+						.doOnNext((socket) -> {
+							if (this.connectSubscribeUrl != null) {
+								socket.connect(this.connectSubscribeUrl);
+								if (this.pubSub) {
+									socket.subscribe(ZMQ.SUBSCRIPTION_ALL);
+								}
+							}
+							else {
+								socket.bind(localPairConnection.get());
+							}
+						})
+						.cache();
+
+		Flux<? extends Message<?>> receiveData =
+				this.subscribeSocket
+						.flatMap((socket) -> {
+							if (this.initialized) {
+								byte[] data = socket.recv(ZMQ.NOBLOCK);
+								if (data != null) {
+									return Mono.just(data);
+								}
+							}
+							return Mono.empty();
+						})
+						.publishOn(Schedulers.parallel())
+						.map(this.messageMapper::toMessage)
+						.doOnError((error) -> logger.error("Error processing ZeroMQ message", error))
+						.repeatWhenEmpty((repeat) ->
+								this.initialized
+										? repeat.delayElements(this.consumeDelay)
+										: repeat)
+						.repeat(() -> this.initialized);
+
+		if (this.pubSub) {
+			receiveData = receiveData.publish()
+					.autoConnect(1, (disposable) -> this.subscriberDataDisposable = disposable);
+		}
+
+		this.subscriberData = receiveData;
+
+	}
+
+	/**
+	 * Configure a connection to the ZeroMQ proxy with the pair of ports over colon
+	 * for proxy frontend and backend sockets. Mutually exclusive with the {@link #setZeroMqProxy(ZeroMqProxy)}.
+	 * @param connectUrl the connection string in format {@code PROTOCOL://HOST:FRONTEND_PORT:BACKEND_PORT},
+	 *                    e.g. {@code tcp://localhost:6001:6002}
+	 */
+	public void setConnectUrl(@Nullable String connectUrl) {
+		if (connectUrl != null) {
```

Comment:
Should this perhaps be handeled by an `Assert`statement? This way it's just silently ignored when someone tries to set the connectUrl to null.

Reply:
I don't think so. This option really can be as a `null`.
Imagine a Spring Boot configuration property when you just propagate it into this setter.
In one environment it can be empty (or missed) and will make this channel in a local mode, e.g. for testing.
In another env it is going to have some reasonable value for connection to the proxy.
Independently of the env your code around this channel configuration is going to be the same.